pipeline/misc/validate_mask_isohits.py

```diff
@@ -0,0 +1,200 @@
+import numpy as np
+import matplotlib.pyplot as plt
+from pixell import enmap
+from soopercool import BBmeta, utils as su, map_utils as mu
+import pymaster as nmt
+import os
+import re
+import argparse
+
+
+X_OFFSET_UNIT = 2  # Plots: Shift in x to avoid overlapping lines
+
+
+def get_filename_tags(dir, base, ext):
+    """
+    Extract all tags from filenames in a directory matching the
+    pattern <base><tag>.<extension>. Tag may be empty ('').
+    """
+    ext = ext.lstrip('.')
+    pattern = re.compile(rf'^{base}(?P<tag>(?:_[^.]*)?)\.{ext}$')
+
+    tags = []
+    for filename in os.listdir(dir):
+        match = pattern.match(filename)
+        if match:
+            tags.append(match.group('tag'))
+
+    return tags
+
+def main(args):
+    """
+    Validate masks generated by get_analysis_mask.py.
+    Use the same param file to ensure paths are correct, with the
+    following requirements:
+
+    Parameters
+    ----------
+    output_directory: Output directory used for get_analysis_mask.py
+
+    general_pars:
+      pix_type: car or hp
+      car_template: Path to car template (if pix_type == car)
+      binning_file: Path to binning file
+
+    mask_validation:
+      num_sims: Number of sims to use
+      sim_id_start: Sim ID to start from
+      unfiltered_map_dir: Directory containing unfiltered maps (sims)
+      unfiltered_map_template: Filename template
+    """
+
+    meta = BBmeta(args.globals)
+    purify_b = args.no_purify
+    fwhm_amin = args.fwhm
+    pix_type = meta.pix_type
+    pure_type = args.puretype
+    conv = args.no_unit_conversion
+    norm_std = args.no_norm_std
+    simdir_unfiltered = meta.mask_validation["unfiltered_map_dir"]
+    unfiltered_map_tmpl = meta.mask_validation["unfiltered_map_template"]
+
+    # Directory tree as structured in get_analysis_mask.py
+    out_dir = meta.output_directory
+    mask_dir = f"{out_dir}/masks"
+    plot_dir = f"{out_dir}/plots/masks"
+    BBmeta.make_dir(plot_dir)
+
+    # Check mask directory for masks
+    mask_basename = "analysis_mask"
```

Comment:
Can we make the default option to analyze only the mask at `meta.mask["analysis_mask"]`, and only if that is null, look for the masks in the mask directory? The reason is that soopercool will by default compute power spectra from the former, and in general `{mask_dir}/analysis_mask.fits` may differ from `meta.mask["analysis_mask"]`.